### PR TITLE
Add EditEntity, Merge and Redirect related diagrams and details

### PR DESCRIPTION
**Pull request by addshore**
In the style of the diagram added in:
https://github.com/wmde/wikidata-wikibase-architecture/pull/16

### CODE REVIEW

File: systems/WikibaseRepo/diagrams/05-api-wbmergeitems.drawio.svg [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            <switch>
+                <foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility">
+                    <div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 108px; height: 1px; padding-top: 55px; margin-left: 391px;">
+                        <div style="box-sizing: border-box; font-size: 0; text-align: center; ">
+                            <div style="display: inline-block; font-size: 15px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; ">
+                                <div style="font-size: 15px">
+                                    ApiBase
+                                </div>
+                                <div style="font-size: 15px">
+                                    (medaiwiki base class)
```
Comment (jakobw) [inline systems/WikibaseRepo/diagrams/05-api-wbmergeitems.drawio.svg:34]:
```suggestion
                                    (MediaWiki base class)
```

**Review by jakobw — CHANGES_REQUESTED** [review]
Found a typo :tada: 

I think these go into too much detail to be super useful. I'd be curious what these would look like if we grouped specific classes into logical components instead. That doesn't need to happen in this PR though since that's part of our homework anyway.

**Review by tzhelyazkova — COMMENTED** [review]
The overall description surrounding those diagrams (maybe in a table format) should also be part of the PR so we don't forget about it.

**Comment by addshore** [comment]
> The overall description surrounding those diagrams (maybe in a table format) should also be part of the PR so we don't forget about it.

I'm going to go and copy how you did it now :)

File: systems/WikibaseRepo/05-Building_Block_View.md [hunk 2]
```diff
@@ -1,3 +1,64 @@
...
+| PermissionManager   | MediaWiki binding | A MediaWiki service class for checking permissions |
+| EntityStore   | Persistence | Storage interface for Entities |
+
+#### EntityRedirectCreationInteractor
+
+![Alt Text](./diagrams/05-interactor-entityredirect.drawio.svg)
+
+|  Building Block | Type/Context  | Responsibility   |
+| ------------ | ------------ | ------------ |
+| ItemRedirectCreationInteractor   | Implementation | Items |
```
Comment (jakobw) [inline systems/WikibaseRepo/05-Building_Block_View.md:57]: I tripped a bit over this implementation being listed here among the building blocks. Listing the implementation among the dependencies seems a bit backwards. OTOH I can see it being useful information, but maybe it shouldn't be in this table.

**Review by jakobw — COMMENTED** [review]
Generally looks ok to me.

Some building block descriptions already appear multiple times which makes me want to extract them somehow to make them reusable instead of copy pasting them. We could use [custom vuepress components](https://vuepress.vuejs.org/guide/using-vue.html#using-components) to transclude the description or we could just link to one central place? This doesn't have to happen in this PR of course.

File: systems/WikibaseRepo/05-Building_Block_View.md (hunk 2 above)
Comment (addshore) [reply systems/WikibaseRepo/05-Building_Block_View.md:85]: I agree, I copied it from the concept board stuff for now to trigger some code review discussion, but I also dont think it shoudl live here.
I'll remove it

**Review by jakobw — APPROVED** [review]
Looks good to me! :rocket:

**Review by Silvan-WMDE — APPROVED** [review]
Looks good for now, but I agree it is probably too much detail.

In terms of usability I like the anchor links from the description tables, but the way the page is structured at the moment, a lot of scrolling is required and thus the existing relations between these diagrams aren't reflected in a good way.

What I am dreaming of are probably interactive SVGs that show components' descriptions as tooltips on hover (instead of tables below the images) and allow jumping to zoom-in diagrams by clicking into building blocks.

**Comment by addshore** [comment]
> What I am dreaming of are probably interactive SVGs that show components' descriptions as tooltips on hover (instead of tables below the images) and allow jumping to zoom-in diagrams by clicking into building blocks.

I think tooltips might be a bit much (even if the dream).
We are free to write plugins etc for vuepress, but I fear that making things too complex will detract from the simplicity.

One of the things I did think about regarding these tables is that the Responsibilitys should probably be here in realtion to the block we are looking at, rather than their general responsobilities?

For example in the merge special page, the responsobility of one of the title services is obviously to lookup titles from entity ids.
However in the context of the block we were looking at it is to link to pages of entities.

**Pull request merged**